boards/qn9080dk/Kconfig

```diff
@@ -0,0 +1,19 @@
+# Copyright (c) 2020 iosabi
+#
+# This file is subject to the terms and conditions of the GNU Lesser
+# General Public License v2.1. See the file LICENSE in the top level
+# directory for more details.
+#
+
+config BOARD
+    default "qn9080dk" if BOARD_QN9080DK
+
+config BOARD_QN9080DK
+    bool
+    default y
+    select BOARD_COMMON_QN9080
+    select CPU_MODEL_QN9080XHN
+
+    # Put defined MCU peripherals here (in alphabetical order)
+    select HAS_PERIPH_GPIO
+    select HAS_PERIPH_GPIO_IRQ
```

Comment:
I was actually wondering that. The CPU always has the periph_gpio module, but whether those are used as gpios or something else depends on the wiring. If the board has no external buttons/gpio/leds etc, would a board with this cpu provide the PERIPH_GPIO feature?

how about other periphs? (uart, i2c, spi)

Reply:
>  If the board has no external buttons/gpio/leds etc, would a board with this cpu provide the PERIPH_GPIO feature?

That's a bit of a philosophical question almost ;)
The thing is that a uart or spi implementation might configure GPIOs (e.g. chip select) and GPIOs don't require any configuration, they are always there.
So it doesn't hurt to always advertise GPIO support by the CPU to keep the board code shorter. It's only used if something `FEATURES_REQUIRED += periph_gpio` anyway.

> how about other periphs? (uart, i2c, spi)

￼Those require some configuration (which pins to use) and they would need to be exposed in order to be useful, so it can only be advertised by the board if they are configured.

Reply:
I'm thinking that an application that implements a blinking LED will set `FEATURES_REQUIRED += periph_gpio` and maybe use them from saul, but it is possible to not compile and include gpio.c in the hello-world application case, even after implementing a uart, i2c and spi drivers. The CS is handled by hardware here too and doesn't involve calling gpio.c code (it does involve setting the mux accordingly, but it is not a GPIO). All of this to say that maybe a board can save some space by not including gpio.c, but it doesn't hurt too much to have it. I moved this to the cpu/.

Reply:
`FEATURES_PROVIDED` is only a hint to the build system that the module*can* be selected. 
It will only be compiled in if it is also in `FEATURES_REQUIRED`